File: server/src/Models/DMToolkitModels/Collections/StartupData.cs
using DMToolkit.Models.Definitions;
using DMToolkit.Models.Entities;

namespace DMToolkit.Models.Collections;

public class StartupData
{
    public List<AbilityScoreDefinition> AbilityScoreDefinitions { get; set; } = new();
    public List<BackgroundDefinition> BackgroundDefinitions { get; set; } = new();
    public List<CharacterClassDefinition> CharacterClassDefinitions { get; set; } = new();
    public List<FeatDefinition> FeatDefinitions { get; set; } = new();
    public List<SpeciesDefinition> SpeciesDefinitions { get; set; } = new();
    public List<Effect> Effects { get; set; } = new();
    public List<School> Schools { get; set; } = new();
    public List<Spell> Spells { get; set; } = new();
}

File: web-client/src/lib/types/dm-tool-types/collections/startupData.ts
import type { AbilityScoreDefinition, AbilityScoreDefinitionDto } from "../definitions/abilityScoreDefinition";
import type { BackgroundDefinition, BackgroundDefinitionDto } from "../definitions/backgroundDefinition";
import type { CharacterClassDefinition, CharacterClassDefinitionDto } from "../definitions/characterClassDefinition";
import type { FeatDefinition, FeatDefinitionDto } from "../definitions/featDefinition";
import type { SpeciesDefinition, SpeciesDefinitionDto } from "../definitions/speciesDefinition";
import type { Effect, EffectDto } from "../entities/effect";
import type { School, SchoolDto } from "../entities/school";
import type { Spell, SpellDto } from "../entities/spell";
import type { ItemDefinitionBase, ItemDefinitionBaseDto } from "../items/bases/itemDefinitionBase";

export interface StartupData {
    abilityScoreDefinitions: AbilityScoreDefinition[];
    backgroundDefinitions: BackgroundDefinition[];
    characterClassDefinitions: CharacterClassDefinition[];
    featDefinitions: FeatDefinition[];
    speciesDefinitions: SpeciesDefinition[];
    effects: Effect[];
    itemDefinitionBases: ItemDefinitionBase[];
    schools: School[];
    spells: Spell[];
}

export interface StartupDataDto {
    abilityScoreDefinitions: AbilityScoreDefinitionDto[];
    backgroundDefinitions: BackgroundDefinitionDto[];
    characterClassDefinitions: CharacterClassDefinitionDto[];
    featDefinitions: FeatDefinitionDto[];
    speciesDefinitions: SpeciesDefinitionDto[];
    effects: EffectDto[];
    itemDefinitionBases: ItemDefinitionBaseDto[];
    schools: SchoolDto[];
    spells: SpellDto[];
}